scipy/linalg/decomp_cholesky.py

```diff
@@ -2,26 +2,33 @@
 
 from __future__ import division, print_function, absolute_import
 
-from numpy import asarray_chkfinite, asarray
+from numpy import asarray_chkfinite, asarray, atleast_2d
 
 # Local imports
 from .misc import LinAlgError, _datacopied
 from .lapack import get_lapack_funcs
 
 __all__ = ['cholesky', 'cho_factor', 'cho_solve', 'cholesky_banded',
-            'cho_solve_banded']
+           'cho_solve_banded']
 
 
 def _cholesky(a, lower=False, overwrite_a=False, clean=True,
-                check_finite=True):
+              check_finite=True):
     """Common code for cholesky() and cho_factor()."""
 
-    if check_finite:
-        a1 = asarray_chkfinite(a)
-    else:
-        a1 = asarray(a)
-    if len(a1.shape) != 2 or a1.shape[0] != a1.shape[1]:
-        raise ValueError('expected square matrix')
+    a1 = asarray_chkfinite(a) if check_finite else asarray(a)
+
+    # Quick return for empty arrays
+    if a1.size == 0:
+        return a, lower
```

Comment:
The error message about shape and dimension is actually correct. "Empty arrays" are not a thing in themselves, but they are useful to avoid the need to special-case handling of `n==0`. Here, the special case is `0x0` matrices. `array([])` and `array([[[]]])` are not `0x0` matrices --- it is only consistent to treat them as matrices if you generally upcast 1d arrays and scalars to 2d matrices, which I believe is not done here.